File: lagom/examples/bc1-to-bc2/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/InventoryService.scala
package com.apuex.sales.mapping.bc1ToBc2

import javax.inject._

import akka.{Done, NotUsed}
import akka.stream.scaladsl.Source
import com.lightbend.lagom.scaladsl.api.{Descriptor, Service, ServiceCall}
       
trait InventoryService extends Service {
  /**
    * Subscribe from event stream with offset.
    *
    * @see [[akka.persistence.query.Offset]]
    * @see [[akka.persistence.query.TimeBasedUUID]]
    * @param offset timed-uuid specifies start position
    * @return
    */
  def events(offset: Option[String]): ServiceCall[Source[String, NotUsed], Source[String, NotUsed]]
         
  def reduce(): ServiceCall[ReduceStorageCmd, Done]

  
  override final def descriptor: Descriptor = {
    import Service._
  
    named("inventory")
      .withCalls(
        pathCall("/api/events", events _)
      ).withAutoAcl(true)
  }
       
}

File: lagom/examples/bc1-to-bc2/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/InventoryService.java
package com.apuex.sales.mapping.bc1ToBc2;

import akka.Done;
import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import java.util.Optional;

import com.lightbend.lagom.javadsl.api.Service;
import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

public interface InventoryService extends Service {


  ServiceCall<ReduceStorageCmd, Done> reduce();

  /**
   * Subscribe from event stream with offset.
   *
   * @param offset timed-uuid or sequence specifies start position
   * @return
   */
  ServiceCall<akka.stream.javadsl.Source<String, NotUsed>, akka.stream.javadsl.Source<String, NotUsed>> events(Optional<String> offset);

  @Override
  default Descriptor descriptor() {
    return named("inventory")
        .withCalls(
            pathCall("/api/events?offset", this::events),
            pathCall("/api/reduce-storage", this::reduce)
        ).withAutoAcl(true);
  }
}
